### PR TITLE
feat: Handle Branch Deeplinks from Braze Push Notification

### PR DESCRIPTION
**Pull request by HamzaIsrar12**
To manage branch deeplinks arriving in Braze push notifications

https://github.com/openedx/openedx-app-android/assets/71447999/dc93a742-9015-4b98-baac-ab396275fbb7

### References
- [Braze Deeplink Handler](https://www.braze.com/docs/developer_guide/platform_integration_guides/android/advanced_use_cases/deep_linking/#universal-deep-link-delegate)

### CODE REVIEW

**Review by dixidroid — DISMISSED** [review]

File: app/src/main/java/org/openedx/app/deeplink/BrazeBranchDeeplinkHandler.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package org.openedx.app.deeplink
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import com.braze.ui.BrazeDeeplinkHandler
+import com.braze.ui.actions.UriAction
+import org.openedx.app.AppActivity
+
+class BrazeBranchDeeplinkHandler : BrazeDeeplinkHandler() {
```
Comment (omerhabib26) [inline app/src/main/java/org/openedx/app/deeplink/BrazeBranchDeeplinkHandler.kt:10]: Please update the name with `BranchBrazeDeeplinkHandler`

**Review by omerhabib26 — APPROVED** [review]
just a minor nit

**Review by omerhabib26 — APPROVED** [review]

**Review by volodymyr-chekyrta — APPROVED** [review]
It's good to merge for now 👍 
In the future, braze will be moved to a plugin

**Comment by HamzaIsrar12** [comment]
@volodymyr-chekyrta What do you mean by a plugin? Can you please provide an example as well if possible? 😄

**Pull request merged**